src/sync/all_forks.rs

```diff
@@ -995,9 +995,36 @@ pub struct AddBlockOccupied<TBl, TRq, TSrc> {
 }
 
 impl<TBl, TRq, TSrc> AddBlockOccupied<TBl, TRq, TSrc> {
+    /// Gives access to the user data of the block.
+    pub fn user_data_mut(&mut self) -> &mut TBl {
+        if self.is_verified {
+            &mut self
+                .inner
+                .inner
+                .chain
+                .non_finalized_block_by_hash(&self.inner.expected_next_hash)
+                .unwrap()
```

Comment:
safe to unwrap?

Reply:
Yeah, a `AddBlockOccupied` struct is only constructed only for blocks that exist.